fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTestDataModel.java

```diff
@@ -0,0 +1,281 @@
+import com.apple.foundationdb.record.test.TestKeySpacePathManagerExtension;
+import com.apple.foundationdb.tuple.Tuple;
+
+import javax.annotation.Nonnull;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Random;
+import java.util.function.Function;
+
+/**
+ * Model for creating a lucene appropriate dataset with various configurations.
+ */
+public class LuceneIndexTestDataModel {
+
+    final boolean isGrouped;
+    final boolean isSynthetic;
+    final boolean primaryKeySegmentIndexEnabled;
+    final int partitionHighWatermark;
+    final Random random;
+    final RandomTextGenerator textGenerator;
+    final Index index;
+    final Function<FDBRecordContext, FDBRecordStore> schemaSetup;
+    final Map<Tuple, Map<Tuple, Tuple>> ids;
+
+    private LuceneIndexTestDataModel(final Builder builder) {
+        random = builder.random;
+        textGenerator = builder.textGenerator;
+        isGrouped = builder.isGrouped;
+        isSynthetic = builder.isSynthetic;
+        primaryKeySegmentIndexEnabled = builder.primaryKeySegmentIndexEnabled;
+        partitionHighWatermark = builder.partitionHighWatermark;
+
+        final Map<String, String> options = new HashMap<>();
+        options.put(LuceneIndexOptions.PRIMARY_KEY_SEGMENT_INDEX_V2_ENABLED, String.valueOf(primaryKeySegmentIndexEnabled));
+        if (partitionHighWatermark > 0) {
+            options.put(LuceneIndexOptions.INDEX_PARTITION_BY_FIELD_NAME, isSynthetic ? "parent.timestamp" : "timestamp");
+            options.put(LuceneIndexOptions.INDEX_PARTITION_HIGH_WATERMARK, String.valueOf(partitionHighWatermark));
+        }
+
+        final RecordMetaDataBuilder metaDataBuilder = LuceneIndexTestDataModel.createBaseMetaDataBuilder();
+        final KeyExpression rootExpression = LuceneIndexTestDataModel.createRootExpression(isGrouped, isSynthetic);
+        index = LuceneIndexTestDataModel.addIndex(isSynthetic, rootExpression, options, metaDataBuilder);
+        final RecordMetaData metadata = metaDataBuilder.build();
+        final StoreBuilderSupplier storeBuilderSupplier = builder.storeBuilderSupplier;
+        final KeySpacePath path = builder.path;
+        schemaSetup = context -> storeBuilderSupplier.get(context, metadata,  path).createOrOpen();
+        ids = new HashMap<>();
+    }
+
+    @Override
+    public String toString() {
+        return "LuceneIndexDataModel{" +
+                "isGrouped=" + isGrouped +
+                ", isSynthetic=" + isSynthetic +
+                ", primaryKeySegmentIndexEnabled=" + primaryKeySegmentIndexEnabled +
+                ", partitionHighWatermark=" + partitionHighWatermark +
+                '}';
+    }
+
+    public void deleteRecord(final FDBRecordContext context, final Tuple primaryKey) {
+        FDBRecordStore recordStore = Objects.requireNonNull(schemaSetup.apply(context));
+        recordStore.getIndexDeferredMaintenanceControl().setAutoMergeDuringCommit(false);
```

Comment:
Should that be done at store creation?

Reply:
Yes, it can. I didn't do that originally because it didn't do it in `schemaSetup` in `LuceneIndexMaintenanceTest`, so it is now redundant in the one place that actually builds an instance of this class, but that's harmless.